### PR TITLE
[#3811] improve(doc): Update and Improve Contributing Guidelines

### PR DESCRIPTION
**Pull request by LanceHsun**
### What changes were proposed in this pull request?

This PR includes the following changes:
1. **Updated Table of Contents**: Added new sections for IntelliJ and VS Code setup.
2. **Added IntelliJ Setup Instructions**: Detailed steps for setting up IntelliJ on Windows using WSL.
3. **Added VS Code Setup Instructions**: Detailed steps for setting up VS Code on Windows using WSL.
4. **Improved Content Structure**: Corrected and enhanced the content structure for better readability and usability.
5. **Removed Unnecessary Blank Lines**: Cleaned up the formatting for consistency.
6. **Added References to Other Files**: Added links to other relevant documentation files (e.g., `README.md`, `GOVERNANCE.md`, `ROADMAP.md`) to guide contributors for better project understanding.
7. **Handling Memory Issues in WSL**: Include steps to manage memory usage in WSL by creating a `.wslconfig` file.

### Why are the changes needed?

The changes are needed to:
1. Provide comprehensive and clear instructions for setting up the development environment using IntelliJ and VS Code on Windows.
2. Improve the documentation structure and readability, making it easier for new contributors to follow.
3. Ensure all necessary steps are documented to help contributors set up their development environment correctly, which can reduce setup-related issues and maintain productivity.
4. Maintain consistency and clarity in the documentation to enhance the overall contributor experience.
5. Provide references to other key documentation files to help contributors understand the project context better.

### Does this PR introduce _any_ user-facing change?

No, this PR does not introduce any user-facing changes. It only updates the documentation for contributors.

### How was this patch tested?

The changes were reviewed for clarity, accuracy, and completeness. Since this is a documentation update, no code changes or tests are applicable. The updated documentation was visually inspected to ensure it renders correctly.

### Related Issue

This PR addresses issue #3811.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,114 +28,136 @@ Before you get started, please read and follow these guidelines to ensure a smoo
...
+   ```
+   This command will open the current WSL directory in VSCode on Windows. If you haven't added `code` to your path, follow these steps:
+   
+   - Open VSCode on Windows.
+   - Press `Ctrl+Shift+P` to open the command palette.
+   - Type and select `Shell Command: Install 'code' command in PATH`.
+3. **Ensure Remote - WSL is Active**
+   When VSCode opens, you should see a green bottom-left corner indicating that VSCode is connected to WSL. If it isn't, click on the green area and select `Remote-WSL: New Window` or `Remote-WSL: Reopen Folder in WSL`.
+4. **Edit and Develop Your Project**
+   You can now edit and develop your project files in VSCode as if they were local files. The Remote - WSL extension seamlessly bridges the file system between Windows and WSL.
```
Comment (jtao1) [inline CONTRIBUTING.md:104]: I ran into a memory issue when using WSL and I was able to find a solution for it. 

```
### Managing WSL
WSL uses a lot of memory so it is recommended to create a `.wslconfig` file that allows memory limits to be put in place.

1. **Shut down WSL**
If your WSL is open you can shut it down in Windows PowerShell using the following command: `wsl --shutdown`

2. **Navigate to user folder**
Open up file explorer and navigate to `C:\Users\<your-username>`. 

3. **Create the `.wslconfig` file**
Open up notepad or another text editor and input the following:

`[wsl2]
memory=4GB   # Limits VM memory in WSL 2 up to 4GB
processors=4 # Makes the WSL 2 VM use two virtual processors`

#The memory and processor usage can be changed depending on your system's hardware.


4. **Saving the file**
Save the file as `".wslconfig"`. Be sure to include the quotes to let Windows know that this isn't a text file.

```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,114 +28,136 @@ Before you get started, please read and follow these guidelines to ensure a smoo
...
 For any bugs or new code please add unit tests to provide coverage of the code. The project may not accept code without unit tests.
 
 All text files should use macOS/unix style line endings (LF) not windows style line endings (CRLF).
-
 ## Community and communication
-
 Join the [community discourse group](https://gravitino.discourse.group) to discuss ideas and seek help. You are also encouraged to use GitHub discussions and follow Datastrato on social media to stay updated on project news.
-
 ## License
-
```
Comment (jerryshao) [inline CONTRIBUTING.md:140]: Hi @LanceHsun please avoid deleting the necessary blank line. What I mean previously is that only **one** blank line is enough, but the blank line is necessary, don't delete it.

**Comment by jerryshao** [comment]
Also, please create a related issue and update the PR title like other PRs.

**Comment by LanceHsun** [comment]
> Also, please create a related issue and update the PR title like other PRs.

ok, done!

File: CONTRIBUTING.md (hunk 1 above)
Comment (LanceHsun) [reply CONTRIBUTING.md:123]: > I ran into a memory issue when using WSL and I was able to find a solution for it. I wrote up the steps I took to resolve my issue.
> 
> ```
> ### Managing WSL
> WSL uses a lot of memory so it is recommended to create a `.wslconfig` file that allows memory limits to be put in place.
> 
> 1. **Shut down WSL**
> If your WSL is open you can shut it down in Windows PowerShell using the following command: `wsl --shutdown`
> 
> 2. **Navigate to user folder**
> Open up file explorer and navigate to `C:\Users\<your-username>`. 
> 
> 3. **Create the `.wslconfig` file**
> Open up notepad or another text editor and input the following:
> 
> `[wsl2]
> memory=4GB   # Limits VM memory in WSL 2 up to 4GB
> processors=4 # Makes the WSL 2 VM use two virtual processors`
> 
> #The memory and processor usage can be changed depending on your system's hardware.
> 
> 
> 4. **Saving the file**
> Save the file as `".wslconfig"`. Be sure to include the quotes to let Windows know that this isn't a text file.
> ```

ok, already updated it into the content

File: CONTRIBUTING.md (hunk 2 above)
Comment (LanceHsun) [reply CONTRIBUTING.md]: ok, done!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -48,46 +51,131 @@ Once you have cloned the [GitHub repository](https://github.com/datastrato/gravi
...
 If you have ideas for enhancements or new features, feel free to create an issue to discuss them. Gravitino welcomes suggestions and provides prompt feedback on their feasibility and relevance.
 
-### Good first issues
-
-If you are new to open source or can't find something to work on check out the [Good First Issues list](https://github.com/datastrato/gravitino/contribute).
-
-### Working on issues
+### Good First Issues
+If you are new to open source or can't find something to work on, check out the [Good First Issues list](https://github.com/datastrato/gravitino/contribute).
```
Comment (jerryshao) [inline CONTRIBUTING.md:172]: Again, you should not change this. You should leave blank line before and after for each paragraph. Please recheck all your changes and update the doc.

Comment (LanceHsun) [reply CONTRIBUTING.md:214]: Got it, done!

**Review by jerryshao — APPROVED** [review]
LGTM. Thanks @LanceHsun for your contribution.

**Pull request merged**